src/test/java/model/base/BasePage.java

```diff
@@ -4,14 +4,41 @@
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 
-public abstract class BasePage<EditPageType extends BaseEditPage<?>, ViewPageType extends BaseViewPage<?>> extends BaseModel {
+import java.util.List;
+
+public abstract class BasePage<EditPageType extends BaseEditPage<?>, ViewPageType extends BaseViewPage<?>, MainPageType> extends BaseModel {
```

Comment:
1. Если есть желание добавить саму эту страницу, то было бы логичней добавить ее первой среди обобщенных типов: _**BasePage<MainPageType, EditPageType extends BaseEditPage<?>, ViewPageType extends BaseViewPage<?>>**_
2. логично было бы защитить новый тип, поставить на него фильтр через **extends**: _**BasePage<MainPageType extends BasePage<?, ?, ?>, EditPageType extends BaseEditPage<?>, ViewPageType extends BaseViewPage<?>>**_
3. Знаки вопорсов в обобщенных типах Edit и View (и Main) говорят о том, что мы не знаем какой тип главной страницы будет, но с ее добавлением эта неизвестность исчезает: 
```suggestion
public abstract class BasePage<MainPageType extends BasePage<MainPageType, EditPageType, ViewPageType>, EditPageType extends BaseEditPage<MainPageType>, ViewPageType extends BaseViewPage<MainPageType>> extends BaseModel {